eng/pipelines/templates/steps/cache-maven-repository.yml

```diff
@@ -0,0 +1,281 @@
+steps:
+- script: |
+    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]%USERPROFILE%\.m2\repository"
+  condition: and(eq(variables['Agent.OS'], 'Windows_NT'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['EnableCache'], 'true'))
+  displayName: Detecting Maven repository on Windows
+- script: |
+    echo "##vso[task.setvariable variable=Maven.RepositoryPath;]$HOME/.m2/repository"
+  condition: and(ne(variables['Agent.OS'], 'Windows_NT'), ne(variables['Build.Reason'], 'PullRequest'), eq(variables['EnableCache'], 'true'))
+  displayName: Detecting Maven repository on Linux and macOS
+- task: CacheBeta@0
+  inputs:
+    key: |
+      JobName-$(Agent.JobName)
+      CacheSalt-$(CacheSalt)
```

Comment:
Where does CacheSalt come from?

Reply:
Its a variable on the pipeline. Until they have a cache invalidation switch I've included it here so we can just update a variable on the pipeline definition to invalidate it forcibly.